content-services/content-sources-backend
# Fixes 4665: improve error responses for upload apis

## Summary

This fixes the error responses for the upload APIs

## Testing steps

1. Create an upload with a size of 0 or less. You should see this error in the response:
```
{
    "errors": [
        {
            "status": 400,
            "title": "error creating upload",
            "detail": "upload size must be greater than 0"
        }
    ]
}
```
2. Create an upload with a size greater than 0 and upload a chunk with the wrong Content-Range start and end bytes. You should see this error in the response:
```
{
    "errors": [
        {
            "status": 400,
            "title": "error uploading chunk",
            "detail": "uploading file chunk: 400 Bad Request: {\"non_field_errors\":[\"Chunk size does not match content range.\"]}"
        }
    ]
}
```
3. Try to upload a chunk without a file in the request. You should see a 400 response code instead of a 500
